build.cake

```diff
@@ -0,0 +1,268 @@
+var INSTALL_DIR = PROJECT_DIR + "src/NUnitTestAdapterInstall/";
+var DEMO_DIR = PROJECT_DIR + "demo/";
+var PACKAGE_DIR = PROJECT_DIR + "package/";
+var PACKAGE_IMAGE_DIR = PACKAGE_DIR + packageName + "/";
+var TOOLS_DIR = PROJECT_DIR + "tools/";
+
+// TODO: Consolidate in one directory if possible
+var ADAPTER_BIN_DIR = ADAPTER_DIR + "bin/" + configuration + "/";
+var TEST_BIN_DIR = TEST_DIR + "bin/" + configuration + "/";
+var INSTALL_BIN_DIR = INSTALL_DIR + "bin/" + configuration + "/";
+var DEMO_BIN_DIR = DEMO_DIR + "NUnitTestDemo/bin/" + configuration + "/";
+
+// Solutions
+var ADAPTER_SOLUTION = PROJECT_DIR + "NUnit3TestAdapter.sln";
+var DEMO_SOLUTION = DEMO_DIR + "NUnit3TestDemo.sln";
+
+// Test Runners
+var NUNIT3_CONSOLE = TOOLS_DIR + "NUnit.ConsoleRunner/tools/nunit3-console.exe";
+
+// Test Assemblies
+var ADAPTER_TESTS = TEST_BIN_DIR + "NUnit.VisualStudio.TestAdapter.Tests.dll";
+var DEMO_TESTS = DEMO_BIN_DIR + "NUnit3TestDemo.dll";
+
+// Packages
+var ZIP_PACKAGE = PACKAGE_DIR + "NUnit3TestAdapter-" + packageVersion + ".zip";
+
+// Custom settings for VSTest
+var VSTestCustomSettings = new VSTestSettings()
+{
+	ArgumentCustomization = args => args.Append("/TestAdapterPath:" + ADAPTER_BIN_DIR)
+};
+
+//////////////////////////////////////////////////////////////////////
+// CLEAN
+//////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+    .Does(() =>
+{
+    CleanDirectory(ADAPTER_BIN_DIR);
+    CleanDirectory(TEST_BIN_DIR);
+    CleanDirectory(INSTALL_BIN_DIR);
+	CleanDirectory(DEMO_BIN_DIR);
+});
+
+
+//////////////////////////////////////////////////////////////////////
+// INITIALIZE FOR BUILD
+//////////////////////////////////////////////////////////////////////
+
+Task("InitializeBuild")
+    .Does(() =>
+{
+    NuGetRestore(ADAPTER_SOLUTION);
+	NuGetRestore(DEMO_SOLUTION);
+
+	if (BuildSystem.IsRunningOnAppVeyor)
+	{
+		var tag = AppVeyor.Environment.Repository.Tag;
+
+		if (tag.IsTag)
+		{
+			packageVersion = tag.Name;
+		}
+		else
+		{
+			var buildNumber = AppVeyor.Environment.Build.Number;
+			packageVersion = version + "-CI-" + buildNumber + dbgSuffix;
+			if (AppVeyor.Environment.PullRequest.IsPullRequest)
+				packageVersion += "-PR-" + AppVeyor.Environment.PullRequest.Number;
+			else
+				packageVersion += "-" + AppVeyor.Environment.Repository.Branch;
+		}
+
+		AppVeyor.UpdateBuildVersion(packageVersion);
+	}
+});
+
+//////////////////////////////////////////////////////////////////////
+// BUILD
+//////////////////////////////////////////////////////////////////////
+
+Task("Build")
+    .IsDependentOn("InitializeBuild")
+    .Does(() =>
+    {
+		BuildSolution(ADAPTER_SOLUTION, configuration);
+		BuildSolution(DEMO_SOLUTION, configuration);
+    });
+
+//////////////////////////////////////////////////////////////////////
+// TEST
+//////////////////////////////////////////////////////////////////////
+
+Task("TestAdapterUsingConsole")
+	.IsDependentOn("Build")
+	.Does(() =>
+	{
```

Comment:
You're right. I'll put in a throw for this one. For the VSTest call, it does throw on failure.

As far as logging goes, it's less useful here because the VSTest tests don't return a meaningful return code as NUnit does - it's either 0 or 1 - and the last "test" of the three is an expected failure.